src/test/java/io/vertx/core/net/NetTest.java

```diff
@@ -3333,6 +3361,102 @@ public void testServerNetSocketShouldBeClosedWhenTheClosedHandlerIsCalled() thro
     await();
   }
 
+  @Test
+  public void testSslHandshakeTimeoutInClient() throws Exception {
+    server.close();
+    client.close();
+
+    NetServerOptions serverOptions = new NetServerOptions().setSsl(true)
+      .setKeyStoreOptions(Cert.SERVER_JKS.get());
+    server = vertx.createNetServer(serverOptions);
+
+    NetClientOptions clientOptions = new NetClientOptions().setSsl(true)
+      .setTrustAll(true)
+      // set 1ms to produce timeout
+      .setSslHandshakeTimeout(1);
```

Comment:
I think we want a larger timeout and test with a regular NetServer (without SSL) that does nothing instead.

Reply:
that will provide a more reliable test

Reply:
by larger : 200ms is fine